### PR TITLE
Add `autocorrect-node` and configure that run lint before git commit.

### PR DESCRIPTION
**Pull request by huacnlee**
image

This will just be checking the changed files before git commit.

The `autocorrect` command provided by `autocorrect-node` NPM, there are no other binary need to install. 

It also will follow the `.autocorrectignore` and `.autocorrectrc` rules.

Ref: #4191

@yin1999 @queengooborg

### CODE REVIEW

**Comment by huacnlee** [comment]
image


Fail example

**Comment by huacnlee** [comment]
image

Hi @yin1999, need you do this:

> reviewdog: This GitHub token doesn't have write permission of Review API [1].

## ReviewDog setup guide

`github-pr-review` reporter reports results to GitHub PullRequest review comments using [GitHub Personal API Access Token](https://github.com/settings/tokens).

- Go to https://github.com/settings/tokens and generate new API token.
- Check `public_repo` for public repositories.
- Add the token into this repo [Actions / Secrets](https://github.com/mdn/translated-content/settings/secrets/actions/new) named as `REVIEWDOG_GITHUB_API_TOKEN`.
- Retry the **AutoCorrect Lint` CI again.

After that, we will get the review suggestion like this:

image

**Comment by yin1999** [comment]
Thanks for the contribution, we are still trying to get a suitable way to provide PR lint suggestions, I will let you know as soon as it is cleared :)

**Comment by yin1999** [comment]
I've setup [pr-check-lint](https://github.com/mdn/translated-content/blob/18ce6ef16835a8549ae433fd97ff2e4dcebbbf5c/.github/workflows/pr-check-lint_content.yml) workflow. Could you make the corresponding changes if you want to use reviewdog.

And could you add this to [`fix:md`](https://github.com/mdn/translated-content/blob/18ce6ef16835a8549ae433fd97ff2e4dcebbbf5c/package.json#L19) and [`lint:md`](https://github.com/mdn/translated-content/blob/18ce6ef16835a8549ae433fd97ff2e4dcebbbf5c/package.json#L22).

So we can notice the broken changes, when this tool is upgraded.

**Comment by yin1999** [comment]
Hey @huacnlee. Do you need any help on this :)

**Comment by huacnlee** [comment]
@yin1999 I have fixed the merge conflicts.

**Comment by yin1999** [comment]
As there are [security limitation on GitHub Actions](https://github.com/mdn/content/pull/28240#issuecomment-1662040586). Could you merge the workflow here into [pr-check-lint_content](https://github.com/mdn/translated-content/blob/935aaa7f6d060e0089b00e0cfd59d651eabd09d2/.github/workflows/pr-check-lint_content.yml#L90-L124), or do the same changes as what we made in this workflow - spreading into two jobs and use [`pull_request_target`](https://docs.github.com/actions/using-workflows/events-that-trigger-workflows#pull_request_target) event instead.

File: .autocorrectignore
Comment (yin1999) [inline .autocorrectignore:6]: I'm not sure could we ignore all path except the sub dir for zh-CN by this (as we are now only enable this for zh-CN), so we can simplify the script below

```suggestion
/*
```

File: package.json
Comment (yin1999) [inline package.json:19]:
```suggestion
    "fix:md": "markdownlint-cli2 --fix \"**/*.md\" && node ./scripts/check-url-locale.js --fix files && yarn autocorrect --fix . && prettier -w \"**/*.md\"",
```

File: package.json
Comment (yin1999) [inline package.json:22]:
```suggestion
    "lint:md": "markdownlint-cli2 \"**/*.md\" && node ./scripts/check-url-locale.js files && yarn autocorrect --lint . && prettier -c \"**/*.md\"",
```

**Review by yin1999 — COMMENTED** [review]
@huacnlee I've pushed some commits to merge autocorrect into our system. Could you have a look on this.

And I'm not sure about the ignore rules:

**Comment by huacnlee** [comment]
@yin1999 The `.autocorrectignore` file is use same rules like the `.gitignore`.

**Review by yin1999 — APPROVED** [review]
LGTM. Thanks @huacnlee.

ping @mdn/core-dev for the review from the development team

File: .github/workflows/markdown-lint-fix.yml
Comment (yin1999) [inline .github/workflows/markdown-lint-fix.yml:64]: Using `ls -d` to filter out non-existed dirs

**Comment by yin1999** [comment]
@bsmth Sorry to bother you, this PR has been there for a long time. And most of the content changes come from myself, I think it would be better to have a double check from other people before we merge this. Do you have time to have a look on this :)

**Comment by bsmth** [comment]
> Sorry to bother you, this PR has been there for a long time. And most of the content changes come from myself, I think it would be better to have a double check from other people before we merge this. Do you have time to have a look on this :)

This doesn't bother me at all, please continue to ping me on PRs where I can help review or unblock - that's what I'm here for :) I will have a look today 🙌🏻

File: .github/workflows/autocorrect-lint.yml
Comment (bsmth) [inline .github/workflows/autocorrect-lint.yml:12]: It might be nice to have this documented somewhere in this repo, but not a blocker for this PR.

**Review by bsmth — APPROVED** [review]
No objections from me apart from adding some docs / linking to https://github.com/huacnlee/autocorrect somewhere. A README.md in https://github.com/mdn/translated-content/tree/main/.github/workflows? We can do in a follow-up.

**Comment by bsmth** [comment]
> ping @mdn/core-dev for the review from the development team

It would be great to have a sign off from dev on this, also 👍🏻

**Review by caugner — APPROVED** [review]
LGTM, only affects zh-CN.

**Pull request merged**
